src/AzureIdentityAccessTokenProvider.cs

```diff
@@ -45,21 +46,26 @@ public AzureIdentityAccessTokenProvider(TokenCredential credential, string [] al
             _scopes.Add("https://graph.microsoft.com/.default"); //TODO: init from the request hostname instead so it doesn't block national clouds?
     }
 
-    /// <summary>
-    /// Gets the authorization token for the given target URI.
-    /// </summary>
-    /// <param name="uri">The target <see cref="Uri"/> to get token for</param>
-    /// <param name="cancellationToken">The <see cref="CancellationToken"/> of the task</param>
-    /// <returns>An authorization token string.</returns>
-    public async Task<string> GetAuthorizationTokenAsync(Uri uri, CancellationToken cancellationToken = default)
+    private const string ClaimsKey = "claims";
+
+    /// <inheritdoc/>
+    public async Task<string> GetAuthorizationTokenAsync(Uri uri, Dictionary<string, object> additionalAuthenticationContext = default, CancellationToken cancellationToken = default)
     {
         if(!AllowedHostsValidator.IsUrlHostValid(uri))
             return string.Empty;
 
         if(!uri.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase))
             throw new ArgumentException("Only https is supported");
 
-        var result = await this._credential.GetTokenAsync(new TokenRequestContext(_scopes.ToArray()), cancellationToken); //TODO: we might have to bubble that up for native apps or backend web apps to avoid blocking the UI/getting an exception
+        string decodedClaim = null;
+        if (additionalAuthenticationContext is not null &&
+                    additionalAuthenticationContext.ContainsKey(ClaimsKey) &&
+                    additionalAuthenticationContext[ClaimsKey] is string claims) {
+            var decodedBase64Bytes = Convert.FromBase64String(claims);
```

Comment:
just to double check, this will throw if the string is malformed, are we going to propagate the exception as is or do we need a wrapper exception?

Reply:
I thought about it for a while but couldn't decide either way, do you have a strong opinion either way?

Reply:
I am on the same boat :) I don't have a strong opinion either way. Having a wrapper could pinpoint the failure but the call stack and the FormatException at that line might be enough.